app/models/cohort_list.rb

```diff
@@ -61,6 +80,23 @@ def generate_patients!
     end
   end
 
+  def self.from_registrations(registrations)
+    new.tap do |cohort_list|
+      cohort_list.data =
+        registrations.map do |registration|
+          REGISTRATION_ATTRS.map { registration.send(_1) }
```

Comment:
I don't now why but `send` always feels like a smell to me. Here's a slightly more idiomatic way that avoids it:

```suggestion
          registration.values_at(*REGISTRATION_ATTRS)
```